contracts/receivers/aave-v2-receiver/main.sol

```diff
@@ -126,6 +126,50 @@ contract AaveV2Migrator is MigrateResolver {
     }
 }
 
+contract InstaFlash is AaveV2Migrator {
+
+    using SafeERC20 for IERC20;
+
+    /**
+     * FOR SECURITY PURPOSE
+     * only Smart DEFI Account can access the liquidity pool contract
+     */
+    modifier isDSA {
+        uint64 id = instaList.accountID(msg.sender);
+        require(id != 0, "not-dsa-id");
+        _;
+    }
+
+    function initiateFlashLoan(
+        address[] calldata _tokens,	
+        uint256[] calldata _amounts,	
+        uint _route, // no use of route but just to follow current flashloan pattern
+        bytes calldata data
+    ) external isDSA {	
+        uint _length = _tokens.length;
+        require(_length == _amounts.length, "not-equal-length");
+        uint[] memory iniBal = new uint[](_length);
+        IERC20[] memory _tokenContracts = new IERC20[](_length);
+        for (uint i = 0; i < _length; i++) {
+            _tokenContracts[i] = IERC20(_tokens[i]);
+            iniBal[i] = _tokenContracts[i].balanceOf(address(this));
+            _tokenContracts[i].safeTransfer(msg.sender, _amounts[i]);
+        }
+        CastData memory cd;
+        (cd.dsa, cd.route, cd.tokens, cd.amounts, cd.dsaTargets, cd.dsaData) = abi.decode(
+            data,
+            (address, uint256, address[], uint256[], address[], bytes[])
```

Comment:
```suggestion
            (address, uint256, address[], uint256[], string[], bytes[])
```
Also, in DSAv2 the `cd.dsaTargets` is an `strings[]`. So, we need to change the data type in the decoding logic over here. Will handle the encoding part on the dsa-connect. If v2 then `strings[]`, If v1 then `address[]`

Reply:
Whoops! Updating. Have to also update the data we'll send from our end.